pgjdbc/src/main/java/org/postgresql/core/v2/ProtocolConnectionImpl.java

```diff
@@ -197,6 +209,15 @@ synchronized void addWarning(SQLWarning newWarning) {
 
   synchronized void addNotification(PGNotification notification) {
     notifications.add(notification);
```

Comment:
On 2 June 2016 at 11:21, David Johnston <notifications@github.com> wrote:

> In pgjdbc/src/main/java/org/postgresql/core/v2/ProtocolConnectionImpl.java
> <https://github.com/pgjdbc/pgjdbc/pull/579#discussion_r65560520>:
>
> > @@ -197,6 +209,15 @@ synchronized void addWarning(SQLWarning newWarning) {
> >
> >    synchronized void addNotification(PGNotification notification) {
> >      notifications.add(notification);
>
> Have you considered how the notifications list gets cleared if a user is
> strictly relying upon the listener API?
>


Apparently not.. I guess this should be closed in the close method, and we
will have to deal with this in pools too.

Suggestions ?

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/579/files/229d10adf6686805004491a0d94de076d8e1a3ea#r65560520>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYz9jqY1SdrI9_NhF0X8I3uPvRz52BLks5qHvTpgaJpZM4IsnG2>
> .
>